types/react/ts5.0/index.d.ts

```diff
@@ -230,30 +232,67 @@ declare namespace React {
     type Key = string | number | bigint;
 
     /**
-     * @internal You shouldn't need to use this type since you never see these attributes
-     * inside your component or have to validate them.
+     * @internal The props any component can receive.
+     * You don't have to add this type. All components automatically accept these props.
+     * ```tsx
+     * const Component = () => <div />;
+     * <Component key="one" />
+     * ```
+     *
+     * WARNING: The implementation of a component will never have access to these attributes.
+     * The following example would be incorrect usage because `Component` would never have access to `key`
+     * ```tsx
+     * const Component = (props: React.Attributes) => props.key;
+     * ```
      */
     interface Attributes {
         key?: Key | null | undefined;
     }
+    /**
+     * The props any component accepting refs can receive.
+     * Class components, host components (e.g. `div`) and forwardRef components can receive refs and automatically accept these props.
```

Comment:
Is 'host components' a name used elsewhere?

Reply:
We use "built-in browser components" in the React docs. I think it makes sense to use that term here as well since "built-in renderer specific components" might not hold true.